src/parser/mod.rs

```diff
@@ -2771,6 +2771,27 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// If the current token is the `expected` keyword followed by
+    /// specified tokens, and consume them and returns true.
+    /// Otherwise, no tokens are consumed and returns false.
+    pub fn parse_keyword_with_tokens(&mut self, expected: Keyword, tokens: &[Token]) -> bool {
+        let index = self.index;
+        match self.peek_token().token {
+            Token::Word(w) if expected == w.keyword => {
+                self.next_token();
```

Comment:
FWIW I think it would be possible to call `peek_nth_token` here as well to avoid having to reset index. However, this works fine too

https://docs.rs/sqlparser/latest/sqlparser/parser/struct.Parser.html#method.peek_nth_token

Reply:
Good suggestion. Changed to use `peek_nth_token`.